archive_converter.py

```diff
@@ -13,8 +13,65 @@
 # limitations under the License.
 
 import email
+import os
+from typing import List, Dict
 from setup_gmail import Message
 
+class ArchiveMessageIndex(object):
+    def __init__(self):
+        self._key_to_thread_start : Dict[str, Message] = {} #Maps message subject (str) to starting message of email thread
```

Comment:
I think the comment pushes this line over 80 char, if so, please put the comment on the preceding line.